egov/egov-egi/src/main/java/org/egov/infra/workflow/service/StateService.java

```diff
@@ -72,4 +84,33 @@ public State getStateById(final Long id) {
         return stateRepository.findOne(id);
     }
 
+    @Transactional
+    public State create(final State state) {
+        return stateRepository.save(state);
+    }
+
+    @Transactional
+    public State update(final State state) {
+        return stateRepository.save(state);
+    }
+
+    public Session getSession() {
+        return entityManager.unwrap(Session.class);
+    }
+
+    public List<State> getStates(List<Long> ownerIds, List<String> types, Long userId) {
+        if (ownerIds != null && !ownerIds.isEmpty())
+            return getSession().createCriteria(State.class)
+                    .setFlushMode(FlushMode.MANUAL).setReadOnly(true).setCacheable(true)
+                    .add(Restrictions.in("type", types))
+                    .add(Restrictions.in("ownerPosition.id", ownerIds))
+                    .add(Restrictions.ne("status", StateStatus.ENDED))
+                    .add(Restrictions.not(Restrictions.conjunction().add(Restrictions.eq("status", StateStatus.STARTED))
+                            .add(Restrictions.eq("createdBy.id", userId))))
+                    .addOrder(Order.desc("createdDate"))
+                    .list();
+        else
+            return Collections.emptyList();
+
+    }
```

Comment:
Can you explain the purpose StateService ? since State by itself can't be exist without a business model.